packages/hdwallet-native/src/crypto/isolation/adapters/ed25519.ts

```diff
@@ -0,0 +1,23 @@
+import { Ed25519 } from "../core";
+
+export class Ed25519Adapter {
+  readonly node: Ed25519.Node;
+
+  constructor(node: Ed25519.Node) {
```

Comment:
Note: I chose not to implement non root node derivation logic for solana at this time as we do not need this functionality